### PR TITLE
issue/#149 backend coverage

### LINKED ISSUE #149: Sonarcloud backend is not showing coverage
**Describe the bug**
In sonarcloud, it shows that the backend has 0% coverage which is not the case. It does not seem to pick up clover

**To Reproduce**
1. Go to [sonarcloud](https://sonarcloud.io/dashboard?id=money-tree_BACKEND)
2. You'll see that the coverage is 0

**Expected behavior**
We expect the coverage to reflect the values given in clover

Bug reported by @Alessandro100

### PR DESCRIPTION
**Pull request by wltrf**
# Related Issue
- closes #149 

# Proposed changes
- Add jacoco coverage report

# Additional Info
- Sonarcloud coverage checks for a jacoco coverage report therefore we can't use the clover report for sonarcoud.
- Need to specify java file path starting from root for sonar exclusions.
- Need to specify class file path starting from target/class for jacoco exclusions.

- Changed the default branch on Sonarcloud to dev, as we want to compare new code with dev instead of master when we do pull requests.

# Reviewer(s)
 - @Alessandro100 @razine-bensari

### CODE REVIEW

**Comment by wltrf** [comment]
The sonarcloud bot says 0% coverage, but that is because it only looks at new code, and the pom.xml is a configuration file and therefore no coverage. This [link](https://sonarcloud.io/dashboard?id=money-tree_BACKEND&pullRequest=152) shows the coverage after merge is 65.8%.

File: backend/pom.xml [hunk 1]
```diff
@@ -164,6 +165,31 @@
                 <artifactId>clover-maven-plugin</artifactId>
```
Comment (razine-bensari) [inline backend/pom.xml:165]: I think we can remove openclover since we are using jacoco in sonarcloud and do not need openclover to generate test coverage reports. Minor detail, it can be removed in another PR too

**Review by razine-bensari — APPROVED** [review]
Nice works! thks!!!!

File: backend/pom.xml (hunk 1 above)
Comment (DPigeon) [reply backend/pom.xml:165]: I think we should remove it as it falls under refactoring.

**Comment by DPigeon** [comment]
Nice work!
